pennylane/transforms/broadcast_expand.py

```diff
@@ -90,24 +90,50 @@ def broadcast_expand(tape):
     >>> fn(qml.execute(tapes, qml.device("default.qubit", wires=1), None))
     array([0.98006658, 0.82533561, 0.54030231])
     """
+    # pylint: disable=too-many-branches, no-member
 
     num_tapes = tape.batch_size
     if num_tapes is None:
         raise ValueError("The provided tape is not broadcasted.")
 
-    # Note that these unbatched_params will have shape (#params, num_tapes)
-    unbatched_params = []
-    for op in tape.operations + tape.observables:
-        for j, p in enumerate(op.data):
-            if op.batch_size and qml.math.ndim(p) != op.ndim_params[j]:
-                unbatched_params.append(qml.math.unstack(p))
-            else:
-                unbatched_params.append([p] * num_tapes)
+    new_preps = [[] for _ in range(num_tapes)]
+    new_ops = [[] for _ in range(num_tapes)]
+
+    for prep in tape.prep:
+        # determine if any parameters of the operator are batched
+        if prep.batch_size:
+            for b in range(num_tapes):
+                new_params = tuple(
+                    p if qml.math.ndim(p) == prep.ndim_params[j] else p[b]
+                    for j, p in enumerate(prep.data)
+                )
+                new_prep = qml.ops.functions.bind_new_parameters(prep, new_params)
+                new_preps[b].append(new_prep)
+        else:
+            # no batching in the operator; don't copy
+            for b in range(num_tapes):
+                new_preps[b].append(prep)
```

Comment:
This block may be worth extracting into a private helper function. Something like:
```
def _split_operations(obj_list: List[Operator], num_tapes: int) -> List[Operator]:
```
This can then just be used twice.